src/OrleansCodeGenerator/GrainMethodInvokerGenerator.cs

```diff
@@ -187,31 +187,56 @@ private static MethodDeclarationSyntax GenerateExtensionInvokeMethod(Type grainT
         /// </returns>
         private static MethodDeclarationSyntax GenerateInvokeMethod(Type grainType, MethodInfo invokeMethod)
         {
-            var methodDeclaration = invokeMethod.GetDeclarationSyntax();
             var parameters = invokeMethod.GetParameters();
 
             var grainArgument = parameters[0].Name.ToIdentifierName();
-            var interfaceIdArgument = parameters[1].Name.ToIdentifierName();
-            var methodIdArgument = parameters[2].Name.ToIdentifierName();
-            var argumentsArgument = parameters[3].Name.ToIdentifierName();
+            var requestArgument = parameters[1].Name.ToIdentifierName();
+            
+            // Store the relevant values from the request in local variables.
+            var interfaceIdDeclaration =
```

Comment:
I think the return type of `ToIdentifierName` should be obvious from its name (it's `IdentifierNameSyntax`). I dropped the "Syntax" from the method name because it's superfluous: almost all types in Roslyn AST API end in "Syntax". If you'd like me to change it, I can.